compair/models/comparison.py

```diff
@@ -118,6 +119,27 @@ def _get_new_comparison_pair(cls, course_id, assignment_id, user_id, pairing_alg
             )) \
             .all()
 
+        answers_with_criterion_score = Answer.query \
```

Comment:
You moved `answers_with_criterion_score` and `assignment_criterion_weights` into the if which is good but left the old code behind